cli/azd/pkg/ext/models.go

```diff
@@ -0,0 +1,32 @@
+package ext
+
+type ScriptType string
+type ScriptLocation string
+
+const (
+	ScriptTypeBash       ScriptType     = "bash"
+	ScriptTypePowershell ScriptType     = "powershell"
+	ScriptLocationInline ScriptLocation = "inline"
+	ScriptLocationPath   ScriptLocation = "path"
+)
+
+type ScriptConfig struct {
+	// Internal name of the hook running for a given command
+	Name string `yaml:",omitempty"`
+	// The type of script hook (bash or powershell)
+	Type ScriptType `yaml:"type,omitempty"`
+	// The location of the script hook (file path or inline)
+	Location ScriptLocation `yaml:"location,omitempty"`
+	// When location is `path` a file path must be specified relative to the project or service
+	Path string `yaml:"path,omitempty"`
+	// When location is `inline` a script must be defined inline
+	Script string `yaml:"script,omitempty"`
+	// When set to true will not halt command execution even when a script error occurs.
+	ContinueOnError bool `yaml:"continueOnError,omitempty"`
+	// When set to true will bind the stdin, stdout & stderr to the running console
+	Interactive bool `yaml:"interactive,omitempty"`
+	// When running on windows use this override config
+	Windows *ScriptConfig `yaml:"windows,omitempty"`
+	// When running on linux/macos use this override config
+	Linux *ScriptConfig `yaml:"linux,omitempty"`
```

Comment:
This ends up being recursive. I think we can separate out the script config wrapper from the OS specific config.

Also, the one inconvenience I could see with this approach is if the user wants the same script to run on both systems. They would end up having to duplicating the script.

Speaking also as an end user authoring an `azd` project, I would much rather having a common scripting language and install a common shell across OSes.

Reply:
If you wanted the same script to run on both OSes than you would only configure the top level ScriptConfig and ensure your script is compatible with all your environments.  You would not have to duplicate the config for both windows & linxux.  The OS specific configurations are only for overrides when needed.